### DISCUSSION TITLE
Defautl case class params without `Option`

### DISCUSSION THREAD

**Discussion by ramytanios** [VirtusLab/scala-yaml#276] [Ideas]
Given 
```scala
case class Foo(x: Int, y: Int = 2)
```
I think would be cool to sucessfully decode 
```yaml
x: 10
```
as `y` has already a default value. I am aware that `y: Option[Int]` can do the job, however it would make it easier to work sometimes with fields without the `Option`. `Pureconfig` supports that too.

**Comment by lbialy** [comment]
Yeah, defaults would be good. We'll look into this.

**Reply by ramytanios** [reply]
Amazing thanks
